util/src/main/java/org/killbill/billing/util/entity/dao/EntitySqlDaoTransactionalJdbiWrapper.java

```diff
@@ -81,15 +84,22 @@ public ReturnType inTransaction(final EntitySqlDao<M, E> transactionalSqlDao, fi
      * @return result from the transaction fo type ReturnType
      */
     public <ReturnType> ReturnType execute(final EntitySqlDaoTransactionWrapper<ReturnType> entitySqlDaoTransactionWrapper) {
+        final String debugInfo = logger.isDebugEnabled() ? getDebugInfo() : null;
+
         final Handle handle = dbi.open();
+        logger.debug("DBI handle created, transaction: {}", debugInfo);
```

Comment:
Should we each time check if `logger.isDebugEnabled()` ? 

Also, i am guessing passing `null` in `debugInfo` would work?

Reply:
> Should we each time check if `logger.isDebugEnabled()` ? 

Not really since `debugInfo` is materialized at this point, so there isn't extra work done if `DEBUG` isn't enabled.

> Also, i am guessing passing `null` in `debugInfo` would work?

Yes, although it should never happen (`Logback` will bail early).